src/pages/Scholarships.jsx

```diff
@@ -26,14 +29,55 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
+  const [amountFilterVals, setAmountFilterVals] = useState({
+    minAmount: 0,
+    maxAmount: 0,
+  });
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(() => {
+    const params = queryString.parse(location.search, { parseNumbers: true });
+    const minParam = params.min ?? 0;
+    const maxParam = params.max ?? 0;
+    setAmountFilterVals({
+      minAmount: minParam > 0 ? minParam : 0,
+      maxAmount: maxParam > 0 ? maxParam : 0,
+    });
+
+    const newParams = { ...params };
+    if (!Number.isInteger(minParam) || minParam <= 0) delete newParams.min;
+    if (!Number.isInteger(maxParam) || maxParam <= 0) delete newParams.max;
+    history.push({ search: queryString.stringify(newParams) });
+  }, [location.search, history]);
 
   useEffect(() => {
     // TODO: Create cancellable promises
     Scholarships.list({ sortField, sortDir })
-      .then(setScholarships)
+      .then((results) => {
+        const filterError =
+          amountFilterVals.maxAmount > 0 &&
+          amountFilterVals.maxAmount <= amountFilterVals.minAmount;
+        if (
+          (amountFilterVals.minAmount === 0 &&
+            amountFilterVals.maxAmount === 0) ||
+          filterError
+        )
```

Comment:
With param pruning you can get rid of this too since you'll know the filter values will be valid by the time the component gets a chance to run this effect.

Reply:
The filterError is to check if the filterMin is less than the filterMax. If that's not the case, should we also prune them from the URL? or would it be fine to keep them and show the original set of scholarships